### PR TITLE
remove sort

### PR DESCRIPTION
**Pull request by plally**

### CODE REVIEW

File: lua/ulx/modules/sh/cfc_constraint_checker.lua [hunk 1]
```diff
@@ -128,42 +128,47 @@ local function compileConstraintResults( _, ply, constraintCounts )
...
+    -- Convert constraint data to list and remove players with no constraints if necessary
+    local constraintCountsList = {}
     for _, ply in pairs( targetPlys ) do
-        local constraintCounts = perPlyConstraints[ply]
-        if showPlysWithNoConstraints or constraintCounts.Total > 0 then
-            local block = compileConstraintResults( caller, ply, constraintCounts )
-            table.insert( dataBlocks, block )
+        local plyCounts = perPlyConstraints[ply]
+        if showPlysWithNoConstraints or plyCounts.Total > 0 then
+            table.insert( constraintCountsList, {
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_constraint_checker.lua:150]: modify this code to convert data to a list so its easier to modify and sort before converting to MsgC args.

File: lua/ulx/modules/sh/cfc_constraint_checker.lua [hunk 2]
```diff
@@ -94,7 +94,7 @@ local function countConstraints( plys )
     return perPlyConstraints
 end
 
-local function compileConstraintResults( _, ply, constraintCounts )
+local function getPlyCountsMsgData( ply, constraintCounts )
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_constraint_checker.lua:97]: I think naming it this makes it clear that the purpose of this functions is creating args for MsgC to visualize the data

File: lua/ulx/modules/sh/cfc_constraint_checker.lua [hunk 3]
```diff
@@ -128,42 +128,47 @@ local function compileConstraintResults( _, ply, constraintCounts )
...
-    end
-
-    local finalData = {}
-    for _, block in pairs( dataBlocks ) do table.Add( finalData, block ) end
-
+    -- Create args for MsgC using constraint count list and send to client
+    -- TODO move visualization code clientside
     net.Start( "CFC_ULX_ConstraintResults" )
-    net.WriteTable( finalData )
+        net.WriteTable( getMsgCArgs( constraintCountsList ) )
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_constraint_checker.lua:163]: I moved everything associated with visualizing the data out of this function and into getMsgCArgs. Keeping that code separate from the rest of the logic makes it easier to work with

**Review by MOEJ-The-Good — APPROVED** [review]
I was going to work on this, but you've beaten me to it.
Either way, you've done better than what I can do, I'm not sure if even my solution would top yours.

**Comment by plally** [comment]
Sorry about that, I didnt realize you were working on fixing this when I did it. 

I would take a look at my comments if you havent. It would provide a good explanation behind the reasoning for all of my changes. The main thing is separating the sorting and other logic from the visualization logic

**Pull request merged**